src/controller/case.controller.js

```diff
@@ -27,3 +28,40 @@ export const caseDetailController = async (request, h) => {
   }
   return h.response(result);
 };
+
+export const caseStageController = async (request, h) => {
+  const { nextStage } = request.payload;
+  const { caseId } = request.params;
+
+  const caseRecord = await caseService.getCase(caseId, request.db);
+  if (!caseRecord) {
+    return Boom.notFound(`Case with id: ${caseId} not found`);
+  }
+
+  const previousStage = caseRecord.currentStage;
+
+  const allMandatoryTasksComplete = await caseService.areMandatoryTasksComplete(
+    caseId,
+    request.db
+  );
+  if (!allMandatoryTasksComplete) {
+    return Boom.badRequest(
+      "Not all mandatory tasks in the current stage are complete."
+    );
+  }
+
+  const updatedCase = await caseService.updateCaseStage(
+    caseId,
+    nextStage,
+    request.db
+  );
+
+  await publishEvent("case_stage_updated", {
```

Comment:
can we add the clientRef here as well as we need to utilise it for when approving applications